gateway/handle.go

```diff
@@ -614,3 +614,228 @@ func (gw *Handle) addToWebRequestQ(_ *http.ResponseWriter, req *http.Request, do
+				status := response.GetErrorStatusCode(errorMessage)
+				responseBody := response.GetStatus(errorMessage)
+				gw.logger.Infow("response",
+					"ip", misc.GetIPFromReq(r),
+					"path", r.URL.Path,
+					"status", status,
+					"body", responseBody)
+				gw.stats.NewTaggedStat(
+					"gateway.write_key_failed_requests",
+					stats.CountType,
+					gw.newSourceStatTagsWithReason(arctx, reqType, errorMessage),
+				).Count(1)
+				http.Error(w, responseBody, status)
+				return
+			} // else success
+			responseBody := response.GetStatus(response.Ok)
+			gw.logger.Debugw("response",
+				"ip", misc.GetIPFromReq(r),
+				"path", r.URL.Path,
+				"status", http.StatusOK,
+				"body", responseBody)
+			gw.stats.NewTaggedStat(
+				"gateway.write_key_successful_requests",
+				stats.CountType,
+				gw.newSourceStatTagsWithReason(arctx, reqType, errorMessage),
+			).Count(1)
+			_, _ = w.Write([]byte(responseBody))
+		}()
+		body, err := gw.getPayload(arctx, r, reqType)
+		if err != nil {
+			errorMessage = err.Error()
+			return
+		}
+		if !gjson.ValidBytes(body) {
+			errorMessage = response.InvalidJSON
+			return
+		}
+		gw.requestSizeStat.Observe(float64(len(body)))
+		type jobObject struct {
+			userID string
+			events []map[string]interface{}
+		}
+		var (
+			sourcesJobRunID  = arctx.SourceJobRunID
+			sourcesTaskRunID = arctx.SourceTaskRunID
+			sourceID         = arctx.SourceID
+			workspaceID      = arctx.WorkspaceID
+			userIDHeader     = r.Header.Get("AnonymousId")
+			ipAddr           = misc.GetIPFromReq(r)
+			eventsBatch      = gjson.GetBytes(body, "batch").Array()
+			isUserSuppressed = gw.memoizedIsUserSuppressed()
+			out              = make([]jobObject, 0, len(eventsBatch))
+		)
+		// skipping filling messageID if empty - expected to be done before
+
+		for idx, v := range eventsBatch {
+			toSet, ok := v.Value().(map[string]interface{})
+			if !ok {
+				errorMessage = response.NotRudderEvent
+				gw.stats.NewTaggedStat(
+					"gateway.write_key_failed_events",
+					stats.CountType,
+					gw.newSourceStatTagsWithReason(arctx, reqType, errorMessage),
+				).Count(1)
+				return
+			}
+			anonIDFromReq := strings.TrimSpace(misc.SanitizeUnicode(misc.GetStringifiedData(toSet["anonymousId"])))
+			userIDFromReq := strings.TrimSpace(misc.SanitizeUnicode(misc.GetStringifiedData(toSet["userId"])))
+			// skipping non-identifiable check - allowing all events
+			eventContext, ok := misc.MapLookup(toSet, "context").(map[string]interface{})
+			if ok {
+				if idx == 0 {
+					if v, _ := misc.MapLookup(eventContext, "sources", "job_run_id").(string); v != "" {
+						sourcesJobRunID = v
+					}
+					if v, _ := misc.MapLookup(eventContext, "sources", "task_run_id").(string); v != "" {
+						sourcesTaskRunID = v
+					}
+					// skip calculate sdkVersion - expected to be done already
+				}
+				// skip bot agent check\
+			}
+
+			if isUserSuppressed(workspaceID, userIDFromReq, sourceID) {
+				gw.logger.Infow("suppressed event",
```

Comment:
Let's use structured non-sugared logging whenever possible please (i.e. `Infon`).

Reply:
Same goes with other occurrences in this PR.